### PR TITLE
Adds docker

### PR DESCRIPTION
**Pull request by dbernstein**
This update includes all changes required to run the mill in docker.  

The docker image contains all the binaries and scripts required to run all of the various nodes that comprise the mill.  Which node the docker container is running as depends on the $NODE_TYPE environment variable which can be set to any of the following values:

- sentinel
- audit-worker
- bit-worker
- dup-worker
- storage-stats-worker
- bit-report-worker

To build the docker container:

```
platforms=linux/amd64; docker buildx build  --build-arg VERSION=4.1.0-SNAPSHOT -f docker/Dockerfile --platform ${platforms}  -t duracloud/mill .
```

This is how I'm running  it in duracloud-aws-deployer:
```
#!/bin/bash
millHome=/tmp/mill-home
mkdir -p $millHome

#copy configuration bucket contents to mill home
aws s3 cp --recursive s3://${mill_s3_config_location} $millHome/

instanceId=`ls /var/lib/cloud/instances`

docker run -d -it -v /sys/fs/cgroup/:/sys/fs/cgroup:ro --cap-add SYS_ADMIN -e HOST_NAME="${instance_prefix}-${node_type}-$instanceId" -e DOMAIN=${domain} -e NODE_TYPE="${node_type}" -e AWS_REGION="${aws_region}" -v $millHome:/mill-home  -v /efs:/efs --name=duracloud-mill  ${mill_docker_container}:${mill_version};
```

Notice `-v /sys/fs/cgroup/:/sys/fs/cgroup:ro --cap-add SYS_ADMIN `:

This is necessary to get cron to work on the nodes that require it (namely the sentinel).  It should be  noted that the volume mapping is specific to debian.

### CODE REVIEW

File: common/src/main/java/org/duracloud/mill/util/CommonCommandLineOptions.java [hunk 1]
```diff
@@ -29,5 +30,12 @@ public CommonCommandLineOptions() {
         configFile.setArgName("file");
         configFile.setRequired(true);
         addOption(configFile);
+
+        Option workdir = new Option(CONFIG_FILE_OPTION, "work-dir", true,
```
Comment (fozboz) [inline common/src/main/java/org/duracloud/mill/util/CommonCommandLineOptions.java:34]: Should this be `WORK_DIR_OPTION`?

Comment (dbernstein) [reply common/src/main/java/org/duracloud/mill/util/CommonCommandLineOptions.java:34]: yes.

File: docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
+FROM ubuntu:impish
+
+RUN apt update && apt install openjdk-11-jre-headless wget cron -y
+RUN mkdir /opt/app
+
+RUN wget https://collectors.sumologic.com/rest/download/deb/64 -O SumoCollector.deb && chmod +x SumoCollector.deb
+RUN dpkg -i SumoCollector.deb
+RUN rm SumoCollector.deb
```
Comment (fozboz) [inline docker/Dockerfile:8]: Put all these `RUN` commands in to a single one joined with `&&`. Makes the image about 100MB smaller.

File: docker/build-and-push-to-dockerhub.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
```
Comment (bbranan) [inline docker/build-and-push-to-dockerhub.sh:1]: It doesn't look like this script is being used (the container is being pushed to the github registry in the workflow). Is there a reason to keep this?

File: docker/scripts/bit-report-worker.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+echo "running workman";
+# execute workman
+java -Dlog.level=$LOG_LEVEL -Dduracloud.home=$MILL_HOME -jar /opt/app/workman-${MILL_VERSION}.jar -c "${CONFIG_FILE_PATH}" -q "queue.name.bit-report" --max-threads 4;
```
[the remark this answers is not in the document]
Comment (bbranan) [reply docker/scripts/bit-report-worker.sh:4]: Ha, nevermind. The bit report worker is workman pointed to the bit report queue.

**Review by bbranan — APPROVED** [review]
This looks good to me @dbernstein, sorry it took so long to get to a review. I'd only suggest dropping the dockerhub script (to avoid future confusion) before a merge.

**Comment by dbernstein** [comment]
@bbranan : this should be good to go.

**Pull request merged**